File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
          int maxSum=Integer.MIN_VALUE;
        int globalSum=0;

        for (int x : nums) {
            globalSum=Math.max(globalSum+x,x);
            maxSum=Math.max(maxSum,globalSum);
        }

        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        mxS=nums[0]
        gls=0
        
        for x in nums:
            gls+=x
            mxS=max(mxS,gls)
            gls=max(gls,0)
        return mxS